### ISSUE TITLE
Python: Few bugs in Session Python Plugin

### ISSUE THREAD

**Issue by Kee2u** [microsoft/semantic-kernel#6494]
**Describe the bug**
There were the following issues with uploading files in the code interpreter sessions.
1. In a chat session, if only the local file path was provided but not the data or the remote file path, it would error since all the variables are required for the function. However, in the logic of the upload_file kernel function, these variables were optional. In my internal implementation, I rewrote the function to only use the local file path variable as follows and this resolved the issue:
```python 
    @kernel_function(name="upload_file", description="Uploads a file for the current Session ID")
    async def upload_file(
        self, *, local_file_path: str = None
    ) -> SessionsRemoteFileMetadata:
        """Upload a file to the session pool.
        Args:
            local_file_path (str): The path to the file on the local machine.
        Returns:
            RemoteFileMetadata: The metadata of the uploaded file.
        """
        
        auth_token = await self._ensure_auth_token()
        self.http_client.headers.update(
            {
                "Authorization": f"Bearer {auth_token}",
                USER_AGENT: SESSIONS_USER_AGENT,
            }
        )
   
        with open(local_file_path, 'rb') as file:
            # Prepare the files dictionary for the request
            files = {
                'file': (os.path.basename(local_file_path), file, 'application/octet-stream')
            }
            response = await self.http_client.post(
                url=f"{self.pool_management_endpoint}python/uploadFile?identifier={self.settings.session_id}",
                json={},
                files=files,
            )

        response.raise_for_status()

        response_json = response.json()
        print("Response JSON")
        print(response_json)
        
        return SessionsRemoteFileMetadata.from_dict(response_json)

```
2. The code above resolved the first error but I ran into a second error where the from_dict() function used above was not able to parse response_json.
This is because response_json looks like this:
``` python
{'$id': '1', '$values': [{'$id': '2', 'filename': 'Test2.csv', 'size': 66, 'last_modified_time': '2024-06-03T17:48:46.2672398Z'}]}
```
but the from_dict() function tries to parse filename and file_data as follows which doesnt match the syntax of the json:
``` python
def from_dict(data: dict) -> "SessionsRemoteFileMetadata":
        """Create a RemoteFileMetadata object from a dictionary."""
        return SessionsRemoteFileMetadata(filename=data["filename"], size_in_bytes=data["bytes"])
```
To fix this I implemented the following code changes in my local implementation:
``` python
    @staticmethod
    def from_dict(data: dict) -> "SessionsRemoteFileMetadata":
        """Create a RemoteFileMetadata object from a dictionary."""
        print(data)
        file_data = data['$values'][0]  # Access the first dictionary in the '$values' list
        filename = file_data['filename']
        size_in_bytes = file_data['size']
        return SessionsRemoteFileMetadata(filename=filename, size_in_bytes=size_in_bytes)
```
3. After the above steps I was able to upload a file to the session. However, when I asked to create code to read the csv in the session and analyze it, it was unable to read it. It would error out because it could not find the correct remote file path. The files are uploaded to '/mnt/data/' but this wasnt located anywhere in the code so I added the following line in the from_dict function:

``` python
 filename = '/mnt/data/' + file_data['filename']
```

Full code here:
``` python
def from_dict(data: dict) -> "SessionsRemoteFileMetadata":
        """Create a RemoteFileMetadata object from a dictionary."""
        file_data = data['$values'][0]  # Access the first dictionary in the '$values' list
        filename = '/mnt/data/' + file_data['filename']
        size_in_bytes = file_data['size']
        return SessionsRemoteFileMetadata(filename=filename, size_in_bytes=size_in_bytes)
```

After the steps above, I was able to upload a file and use code interpreter on it. I havent tested the download feature yet but I suspect it will have similar issues 
****
**To Reproduce**
1. Run the following code in a python notebook:
``` python
# Copyright (c) Microsoft. All rights reserved.

import asyncio
import datetime

from azure.core.credentials import AccessToken
from azure.core.exceptions import ClientAuthenticationError
from azure.identity import DefaultAzureCredential

from semantic_kernel.connectors.ai.function_call_behavior import FunctionCallBehavior
from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.azure_chat_prompt_execution_settings import (
    AzureChatPromptExecutionSettings,
)
from semantic_kernel.connectors.ai.open_ai.services.azure_chat_completion import AzureChatCompletion
from semantic_kernel.contents.chat_history import ChatHistory
from semantic_kernel.core_plugins.sessions_python_tool.sessions_python_plugin import (
    SessionsPythonTool,
)
from semantic_kernel.core_plugins.time_plugin import TimePlugin
from semantic_kernel.exceptions.function_exceptions import FunctionExecutionException
from semantic_kernel.functions.kernel_arguments import KernelArguments
from semantic_kernel.kernel import Kernel

from fastapi import FastAPI, WebSocket




from dotenv import load_dotenv

app = FastAPI()  

load_dotenv()

auth_token: AccessToken | None = None

ACA_TOKEN_ENDPOINT = "https://acasessions.io/.default"
#ACA_TOKEN_ENDPOINT = "https://dynamicsessions.io"

async def auth_callback() -> str:
    """Auth callback for the SessionsPythonTool.
    This is a sample auth callback that shows how to use Azure's DefaultAzureCredential
    to get an access token.
    """
    global auth_token
    current_utc_timestamp = int(datetime.datetime.now(datetime.timezone.utc).timestamp())

    if not auth_token or auth_token.expires_on < current_utc_timestamp:
        credential = DefaultAzureCredential()

        try:
            auth_token = credential.get_token(ACA_TOKEN_ENDPOINT)
        except ClientAuthenticationError as cae:
            err_messages = getattr(cae, "messages", [])
            raise FunctionExecutionException(
                f"Failed to retrieve the client auth token with messages: {' '.join(err_messages)}"
            ) from cae

    return auth_token.token


kernel = Kernel()

service_id = "sessions-tool"

chat_service = AzureChatCompletion(
    deployment_name="****",
    service_id=service_id,
    base_url="****",
)
kernel.add_service(chat_service)

sessions_tool = SessionsPythonTool(
    auth_callback=auth_callback,
)

sessions_python_tool = kernel.add_plugin(sessions_tool, "SessionsTool")
kernel.add_plugin(TimePlugin(), "Time")

chat_function = kernel.add_function(
    prompt="{{$chat_history}}{{$user_input}}",
    plugin_name="ChatBot",
    function_name="Chat",
)

req_settings = AzureChatPromptExecutionSettings(service_id=service_id, tool_choice="auto")

filter = {"excluded_plugins": ["ChatBot"]}
req_settings.function_call_behavior = FunctionCallBehavior.EnableFunctions(auto_invoke=True, filters=filter)

arguments = KernelArguments(settings=req_settings)

history = ChatHistory()


async def chat() -> bool:
    try:
        user_input = input("User:> ")
    except KeyboardInterrupt:
        print("\n\nExiting chat...")
        return False
    except EOFError:
        print("\n\nExiting chat...")
        return False

    if user_input == "exit":
        print("\n\nExiting chat...")
        return False
    


    arguments["chat_history"] = history
    arguments["user_input"] = user_input
    
    answer = await kernel.invoke(
        function=chat_function,
        arguments=arguments,
    )
    print(f"Mosscap:> {answer}")
    history.add_user_message(user_input)
    history.add_assistant_message(str(answer))
    return True


async def main() -> None:
    print(
        "Welcome to the chat bot!\
        \n  Type 'exit' to exit.\
        \n  Try a Python code execution question to see the function calling in action (i.e. what is 1+1?)."
    )
    chatting = True


    while chatting:
        chatting = await chat()


if __name__ == "__main__":
    asyncio.run(main())
```
Make sure your .env file has these values
``` python
AZURE_OPENAI_ENDPOINT
AZURE_OPENAI_API_KEY
ACA_POOL_MANAGEMENT_ENDPOINT
```

**Expected behavior**
If you run the code without the changes you should see similar errors as shown above.

**Platform**
 - OS: Windows
 - IDE: VS Code
 - Language :Python
 - Source: semantic-kernel 1.0.3

**Comment by evchaki** [comment]
@moonbox3 - can you take a look at this?

**Comment by moonbox3** [comment]
Hi @Kee2u, thanks for filing the issue and for including the details. It helps a lot.

A few notes:
- It does look like we're out of sync with ACA's APIs. One of the first things I notice is yes, they're returning a list of file upload metadata after uploading a file instead of a dictionary like they used to. We'll update this, and I'll reach out to them to better understand how breaking changes are communicated.
- In our 1.0.4 release today, we've improved the JSON schema building that occurs when a kernel function from method is created. This means that Annotations and Optionals are properly handled and sent to the model as such -- the model is aware of what parameters are optional and which ones are required. As I am testing the plugin right now, I am only getting the requested file path that I gave the model such as: `Please upload `./ans.py` to my code interpreter` and I only get one argument which is `{"local_file_path": "./ans.py"}`.
- I also agree that while building up the remote_file_path, if not included, we need to include `/mnt/data/` with the name we use. Something like:

```
if local_file_path:
    if not remote_file_path:
        remote_file_path = f"/mnt/data/{os.path.basename(local_file_path)}"
    data = open(local_file_path, "rb")  # noqa: SIM115
```

With further testing I see:

```
Mosscap:> The file `ans.py` has been successfully uploaded to the session container at `/mnt/data/ans.py`.
User:> List available files in my sessions container.
Mosscap:> The following file is available in your session container:

- `ans.py` (34 bytes)
```


The fixes are here: #6507

**Comment by Kee2u** [comment]
Hi @moonbox3, 

This is great! The fixes look awesome. Thank you for the quick resolution!
